module3-crud-and-data-models/r1-schema-and-data-models/README.md

```diff
@@ -0,0 +1,99 @@
+## Intoduction to database schemas
+
+A database schema is a blueprint or architecture of how our data will look. It doesn’t hold data itself, but instead describes the shape of the data and how it might relate to other tables or models.
+
+For example, this is a `customer` schema in MongoDB using mongoose.
+
+```js
+const mongoose = require("mongoose");
+
+const customer = new mongoose.Schema({
+  name: {
+    type: String,
+    required: true,
+  },
+  zipcode: {
+    type: Number,
+  },
+});
+
+module.exports = mongoose.model("Customer", customer);
+```
+
+The same `customer` schema in a MySql database.
+
+```sql
+CREATE TABLE customer (
+ id INT AUTO_INCREMENT PRIMARY KEY,
+ name VARCHAR(50) NOT NULL,
+ zipcode INT(15) default NULL,
+)
+
+```
+
+We can see that the schema definition for the `customer` has a `name` which is a type `string` and a `zipcode` that is of type `number`.
+
+## Introduction to CRUD operations
+
+When we are building APIs, we want our models to provide four basic types of functionality. The model must be able to Create, Read, Update, and Delete resources. Computer scientists often refer to these functions by the acronym `CRUD`. A model should have the ability to perform at most these four functions in order to be complete.
+
+### CRUD and REST
+
+### What is REST?
+
+REST is a set of architectural constraints. API developers can implement REST in a variety of ways. When a client request is made via a RESTful API, it transfers a representation of the state of the resource to the requester or endpoint. This information, or representation, is delivered in one of several formats, JSON is the most generally popular file format to use.
+
+### CRUD (CREATE, READ, UPDATE, DELETE)
+
+In a REST environment, CRUD often corresponds to the HTTP methods `POST`, `GET`, `PUT`, and `DELETE`, respectively. These are the fundamental elements of a persistent storage system.
+
+For example, imagine we are working with a system that is keeping track of meals and their corresponding prices for a restaurant. Let’s look at how we would implement CRUD operations.
+
+#### Create
+
+To create resources in a REST environment, we most commonly use the HTTP POST method. POST creates a new resource of the specified resource type.
+Imagine that we are adding a new food item to the stored list of dishes for this restaurant, and the dish objects are stored in a dishes resource.
```

Comment:
you should define what "resource" means somewhere

Reply:
Done in module 1